newIDE/app/src/VariablesList/index.js

```diff
@@ -254,10 +281,50 @@ export default class VariablesList extends React.Component<Props, State> {
     );
   }
 
+  _createInheritedInstanceVariable = (name, value, isStructure) =>{
+    // Takes a variable from the inherited and makes a copy of it to insert in the variables container
+    const { variablesContainer, inheritedVariablesContainer } = this.props;
+    if (inheritedVariablesContainer) {
+      const serializedVariable = serializeToJSObject(
+        inheritedVariablesContainer.get(name)
+      );
+      const newVariable = new gd.Variable();
+      console.log(newVariable)
+
+      if (!isStructure) {
+        newVariable.setString(value)
+      }
+      
+      unserializeFromJSObject(newVariable, serializedVariable);
+      variablesContainer.insert(
+        name,
+        newVariable,
+        variablesContainer.count()//will need to use index?
+      );
+      newVariable.delete();
+    }
+  }
+
   render() {
-    const { variablesContainer } = this.props;
+    const { variablesContainer, inheritedVariablesContainer } = this.props;
     if (!variablesContainer) return null;
 
+    /// map all inherited variables, if they have been passed
+    const containerObjectVariablesTree = inheritedVariablesContainer? mapFor(0, inheritedVariablesContainer.count(), index => {
+      const name = inheritedVariablesContainer.getNameAt(index);
+      if (!variablesContainer.has(name)) { // Show only object variables that have no instance variable
+        const variable = inheritedVariablesContainer.getAt(index);
+        return this._renderVariableAndChildrenRows(
+          name,
+          variable,
+          0,
+          index,
+          undefined
+        );
+      }
+    }):{}
```

Comment:
This should be an empty array: `: []`.
Otherwise `containerObjectVariablesTree` is either an array or an empty object, which can create subtle bugs. Always stick to one type for a variable - make things much easier to reason about.